samcli/lib/telemetry/metric.py

```diff
@@ -148,16 +148,18 @@ def wrapped(*args, **kwargs):
             # Execute the function and capture return value. This is returned back by the wrapper
             # First argument of all commands should be the Context
             return_value = func(*args, **kwargs)
-        except UserException as ex:
+        except (UserException, click.BadOptionUsage, click.BadArgumentUsage, click.BadParameter) as ex:
             # Capture exception information and re-raise it later so we can first send metrics.
             exception = ex
-            exit_code = ex.exit_code
-            if ex.wrapped_from is None:
-                exit_reason = type(ex).__name__
-            else:
+            # TODO (hawflau): review exitcode meaning. We now understand exitcode 1 as errors fixable by users.
+            exit_code = 1
```

Comment:
Why did this change from `ex.exit_code` to `1`?

Reply:
`click.BadOptionUsage` and other click exception caught here come with exitcode=2 by default. In our Telemetry, they were, however, classified as 255 (as we didn't catch any them). Today, we only have 1 and 255 exitcodes in our Telemetry. We should revisit the exitcode meaning someday. But for now, re-classifying them as exitcode 1 seems more correct to me.

Reply:
But why not use whatever actual exit code is thrown, instead of only either 1 or 255?

Reply:
For now I don't want to introduce new exitcode (e.g. 2) into Telemetry, until we have clearer understandings about how we want to use the exitcode. Today, `click.BadOptionUsage` and `click.UsageError` (maybe a couple other user-fixable exceptions as well) go into Telemetry with exitcode 255, which is certainly incorrect. I just want to fix this first.